impl/src/main/java/org/ehcache/impl/internal/store/DefaultStoreProvider.java

```diff
@@ -46,10 +49,22 @@
 public class DefaultStoreProvider implements Store.Provider {
   private static final Logger LOG = LoggerFactory.getLogger(DefaultStoreProvider.class);
 
+  private static final Set<ResourceType> SUPPORTED_RESOURCE_TYPES =
```

Comment:
If we can assume that anything supported by `DefaultStoreProvider` (or its successor) will be in `ResourceType.Core`, then an enum set will work.  I'll tweak this.